### PR TITLE
Bug/met 2523 exectute ghostscript separately

### PR DESCRIPTION
**Pull request by jochen-vermeulen**

### CODE REVIEW

File: metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/TextProcessor.java [hunk 1]
```diff
@@ -75,8 +80,13 @@ public ResourceExtractionResultImpl extractMetadata(Resource resource, String de
...
     final List<Thumbnail> thumbnails;
     if (PDF_MIME_TYPE.equals(detectedMimeType)) {
-      thumbnails = thumbnailGenerator.generateThumbnails(resource.getResourceUrl(),
-          detectedMimeType, resource.getContentFile()).getRight();
+      final Path pdfImage = pdfToImageConverter.convertToPdf(resource.getContentPath());
+      try {
+        thumbnails = thumbnailGenerator.generateThumbnails(resource.getResourceUrl(),
+                PNG_MIME_TYPE, pdfImage.toFile(), true).getRight();
+      } finally {
+        removePdfImageFileSilently(pdfImage);
```
Comment (stzanakis) [inline metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/TextProcessor.java]: Perhaps reuse the method in the pdfToImageConverter?

File: metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/TextProcessor.java [hunk 2]
```diff
@@ -107,6 +118,26 @@ public ResourceExtractionResultImpl extractMetadata(Resource resource, String de
...
+        try {
+          thumbnail.close();
+        } catch (IOException e) {
+          LOGGER.warn("Could not close thumbnail: {}", thumbnail.getTargetName(), e);
+        }
+      }
+    }
+  }
+
+  void removePdfImageFileSilently(Path file) {
```
Comment (stzanakis) [inline metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/TextProcessor.java]: Isn't this duplicate, same code exists in the PdfToImageConverter

File: metis-media-service/src/test/java/eu/europeana/metis/mediaprocessing/extraction/PdfToImageConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            "-sOutputFile=" + outputFile.toString(),
+            "-f" + inputFile.toString());
+
+    // Execute the command
+    assertEquals(expectedCommand,
+            pdfToImageConverter.createPdfConversionCommand(inputFile, outputFile));
+  }
+
+  @Test
+  void testConvertToPdf() throws MediaExtractionException {
```
Comment (stzanakis) [inline metis-media-service/src/test/java/eu/europeana/metis/mediaprocessing/extraction/PdfToImageConverterTest.java]: Observation:
Why are you grouping test cases in one test method?
Shouldn't we be having a method for each case?

Comment (jochen-vermeulen) [reply metis-media-service/src/test/java/eu/europeana/metis/mediaprocessing/extraction/PdfToImageConverterTest.java]: That's one approach... personally I don't like to have to repeat the exact same setup multiple times.

File: metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/TextProcessor.java (hunk 1 above)
Comment (jochen-vermeulen) [reply metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/TextProcessor.java]: Fixed.

File: metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/TextProcessor.java (hunk 2 above)
Comment (jochen-vermeulen) [reply metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/TextProcessor.java]: Fixed.

**Pull request merged**